### PR TITLE
wal/wal.go : improved coverage by testing WAL.Save which causes a WAL…

### PR DESCRIPTION
**Pull request by vimalk78**
completed the TODO:

wal/wal.go // TODO: add a test for this code path when refactoring the tests

files modified:
wal/wal.go : removed the TODO
wal/wal_test.go : added test case TestSaveWithCut

this solves the above TODO

### CODE REVIEW

File: wal/wal_test.go [hunk 1]
```diff
@@ -213,6 +213,39 @@ func TestCut(t *testing.T) {
...
+	defer w.Close()
+
+	state := raftpb.HardState{Term: 1}
+	if err = w.Save(state, nil); err != nil {
+		t.Fatal(err)
+	}
+	bigData := make([]byte, 500)
+	totalSize := 0
+	// set a lower value for SegmentSizeBytes, else the test takes too long to complete
+	restoreLater := SegmentSizeBytes
```
Comment (xiang90) [inline wal/wal_test.go]: let's use defer? 

```go
SegmentSizeBytes = 2 * 1024
defer func() {SegmentSizeBytes = xxx}
```

File: wal/wal_test.go [hunk 2]
```diff
@@ -213,6 +213,39 @@ func TestCut(t *testing.T) {
...
+	SegmentSizeBytes = 2 * 1024
+	var index uint64 = 0
+	for ; totalSize < int(SegmentSizeBytes); index++ {
+		ents := []raftpb.Entry{{Index: index, Term: 1, Data: bigData}}
+		if err = w.Save(state, ents); err != nil {
+			t.Fatal(err)
+		}
+		totalSize += 500
+	}
+	SegmentSizeBytes = restoreLater
```
Comment (xiang90) [inline wal/wal_test.go]: can we check the newly cut wal file is valid?

1. we close the current wal.
2. we check there is one more file created. 
3. reopen the wal dir and verify we can read all entries that we wrote before

**Comment by vimalk78** [comment]
i have done the changes, but something terrible happened while squashing the new commit. 
now this commit is lost in my local branch. after i squashed, the commit id is different.

File: wal/wal_test.go (hunk 2 above)
Comment (vimalk78) [inline wal/wal_test.go]: @xiang90 can you please check

Comment (xiang90) [inline wal/wal_test.go]: Sorry. I do not quite understand what you mean. Do you want me to write the checking code after this pr get merged? Or you have already finished the code and want me to check it? I did not see your updates on this pull request.

Comment (vimalk78) [inline wal/wal_test.go]: i have made two more commits.

Comment (xiang90) [inline wal/wal_test.go]: the two more commits contain no changes. :( I think your git branch is messed up somehow.

Comment (vimalk78) [inline wal/wal_test.go]: ohh, yes, something has got messed up yesterday. i could not bring the changes in the pull request.
shall i delete this branch and PR and make a fresh pr ?

Comment (xiang90) [inline wal/wal_test.go]: ok. please close this and create a new pr. I can help you with git next time. i do not really have time to deal with git right now. :(

Comment (vimalk78) [inline wal/wal_test.go]: thanks a lot :)

Comment (vimalk78) [inline wal/wal_test.go]: closing the PR, created a new PR https://github.com/coreos/etcd/pull/6495

**Pull request closed without merging**